spec/features/posts_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+RSpec.describe 'Post show page', type: :feature do
+  before :each do
+    @user1 = User.create(name: 'Bill', photo: 'https://picsum.photos/300/300', bio: 'Lorem ipsum')
+    @user2 = User.create(name: 'Djill', photo: 'https://picsum.photos/200/400', bio: 'Lorem ipsum')
+    @post1 = Post.create(author: @user1, title: 'Go', text: 'testtest')
+    Comment.create(author: @user1, post: @post1, text: 'testtesttesttest')
+    Comment.create(author: @user1, post: @post1, text: 'testtesttesttest2')
+    Like.create(author: @user1, post: @post1)
+    Like.create(author: @user2, post: @post1)
+    visit user_post_path(@user2, @post1)
+  end
+  it "show the post's title" do
+    expect(page).to have_content(@post1.title)
+  end
+  it 'Show who wrote the post' do
+    expect(page).to have_content(@post1.author.name)
+  end
+  it 'displays the comment count and like count' do
+    expect(page).to have_content('comments:2 likes:2')
+  end
+  it 'Show the post body' do
+    expect(page).to have_content(@post1.text)
+  end
+  it 'Show the username of each commentor' do
+    @post1.comments.each do |comment|
+      expect(page).to have_content(comment.author.name)
+    end
+  end
+  it 'Show the comment each commentor left' do
+    @post1.comments.each do |comment|
+      expect(page).to have_content(comment.text)
+    end
+  end
+end
+
+RSpec.describe 'Posts index ', type: :feature do
```

Comment:
- [Optional] Testing Post index and show pages on the same file may not be a problem. The problem here is that you are writing feature tests(`RSpec.describe 'Posts index ', type: :feature do`) twice on the same file. Each feature test should be written on a separate file to test that particular feature as requested by the previous reviewers. Keenly once more check what you have done in the users_spec.rb file and see the different. There you have a single request spec for both the user index and show page(`RSpec.describe UsersController, type: :request do`)

Reply:
@FrederickMih Thank you sir, I'll restructure it once I finish all the milestones.